src/mmrelay/db_runtime.py

```
@@ -0,0 +1,253 @@
+                    # journal_mode pragma returns the applied mode; ignore result
+                    conn.execute("PRAGMA journal_mode=WAL")
+                conn.execute("PRAGMA foreign_keys=ON")
+                for pragma, value in self._extra_pragmas.items():
+                    # Validate pragma name to prevent injection.
+                    if not re.fullmatch(r"[a-zA-Z_]+", pragma):
+                        raise ValueError(f"Invalid pragma name provided: {pragma}")
+                    # Validate and sanitize value to prevent injection
+                    if isinstance(value, str):
+                        # Allow only alphanumeric, spaces, and common punctuation for string values
+                        if not re.fullmatch(r"[a-zA-Z0-9_\-\s,\.]+", value):
+                            raise ValueError(f"Invalid pragma value provided: {value}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)